src/scripts/main.js

```diff
@@ -1,3 +1,267 @@
 'use strict';
 
-// write your code here
+const theOnlyButton = document.querySelector('.button');
+const board = document.querySelector('tbody');
+const boardSize = 4;
+const pointsCounter = document.querySelector('.game-score');
+const messageForWinner = document.querySelector('.message-win');
+const messageForLooser = document.querySelector('.message-lose');
+const allCells = document.querySelectorAll('td');
+const winnerScore = 2048;
+let score = 0;
+
+theOnlyButton.addEventListener('click', () => {
+  changeButtonClass();
+  hideStartMessage();
+
+  [...allCells].forEach(cell => {
+    cell.innerHTML = '';
+    updateClassList(cell);
+
+    return cell;
+  });
+
+  if (messageForLooser.classList.contains('hidden') === false) {
+    messageForLooser.classList.add('hidden');
+  }
+
+  if (messageForWinner.classList.contains('hidden') === false) {
+    messageForWinner.classList.add('hidden');
+  }
+
+  score = 0;
+  pointsCounter.textContent = score.toString();
+  generateNumber();
+});
+
+function changeButtonClass() {
+  theOnlyButton.textContent = 'Restart';
+  theOnlyButton.classList.remove('start');
+  theOnlyButton.classList.add('restart');
+}
+
+function hideStartMessage() {
+  document.querySelector('.message-start').classList.add('hidden');
+}
+
+function generateNumber() {
+  const isEmptyCell = [...allCells].some(cell => cell.textContent === '');
+
+  if (!isEmptyCell) {
+    messageForLooser.classList.remove('hidden');
+
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const horizontallyCell = Math.floor(Math.random() * boardSize);
+    const verticallyCell = Math.floor(Math.random() * boardSize);
+    const cellForNewNumber = board.rows[horizontallyCell].cells[verticallyCell];
+
+    if (cellForNewNumber.textContent === '') {
+      const newNumber = Math.random() >= 0.9 ? 4 : 2;
+
+      cellForNewNumber.textContent = `${newNumber}`;
+      updateClassList(cellForNewNumber);
+      found = true;
+    }
+  }
+}
+
+document.addEventListener('keyup', (e) => {
```

Comment:
You listen for any key to be up, so that means that if the user clicks ANY key the `hideStartMessage`, `changeButtonClass` functions will be called, as well as `pointsCounter.textContent` will be changed.